### DISCUSSION TITLE
Tabs

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#232] [Component suggestions]
<img width="480" alt="visuele weergave van het tabs component" src="https://github.com/nl-design-system/backlog/assets/545207/82f01898-e521-4a29-b67c-d4f3741bccf2">

We willen graag meer leren over het Tabs component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie? En wat is het responsive gedrag?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1709819541844959)

---

## Naam
Tabs

## Doel
Paneel om onderwerpen te groeperen, waarbij er maar één per keer zichtbaar is, met knoppen om te kiezen welk onderwerp wordt weergegeven.

## Gerelateerde componenten
[Accordion](https://github.com/orgs/nl-design-system/discussions/217)

## Onderzoek

[Tabs used right - Nielsen Norman Group](https://www.nngroup.com/articles/tabs-used-right/)
Artikel waarin wordt beschreven welke design richtlijnen je moet aanhouden bij het ontwerpen van Tabs.

[Tabbed interfaces - Inclusive components](https://inclusive-components.design/tabbed-interfaces/)
Artikel waarin een Tabs component uitvoerig wordt beschreven om zo toegankelijk op te zetten.

[Tabs - BBC GEL](https://bbc.github.io/gel/components/tabs/)
Tab pagina op documentatie website van BBC GEL Design system waarin het tabs component wordt beschreven om toegankelijk in te zetten.

[Don't be afraid of the big long page - Amy Hupe](https://amyhupe.co.uk/articles/dont-be-afraid-of-the-big-long-page/)
Artikel waarin wordt beschreven dat het inzetten van Tabs (of Accordions) niet altijd de oplossing zijn.

**Comment by jeffreylauwers** [comment]
## Amsterdam

- [Storybook](https://amsterdam.github.io/design-system/?path=/docs/components-containers-tabs--docs)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/tabs)
- [Figma](https://www.figma.com/file/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?type=design&node-id=9081-33829&mode=design&t=3RM9VrRKprFqj2Te-4)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-03-21 om 15 22 34](https://github.com/nl-design-system/backlog/assets/545207/481f8bff-5377-4f68-a243-b3386f548889)

Responsive gedrag bij small viewport
![Scherm­afbeelding 2024-03-21 om 15 24 48](https://github.com/nl-design-system/backlog/assets/545207/37bc4c24-c131-4ea8-8901-15dd31870c45)


**Figma**

Tab states
![Scherm­afbeelding 2024-03-21 om 15 25 57](https://github.com/nl-design-system/backlog/assets/545207/9e6145cf-11de-4749-ac0d-aa2c2b07a6e8)

Tabs component
![Scherm­afbeelding 2024-03-21 om 15 26 06](https://github.com/nl-design-system/backlog/assets/545207/47653a6a-a0e7-44f1-b7c8-504c5c984c96)

Responsive gedrag bij small viewport
![Scherm­afbeelding 2024-03-21 om 15 26 20](https://github.com/nl-design-system/backlog/assets/545207/dfd95c37-dc64-428a-b99b-d2509567edde)

![Scherm­afbeelding 2024-03-21 om 15 26 45](https://github.com/nl-design-system/backlog/assets/545207/335970db-2f19-4046-b565-eef3aa1a2530)

Responsive gedrag bij small viewport - Alternatief
![Scherm­afbeelding 2024-03-21 om 15 28 12](https://github.com/nl-design-system/backlog/assets/545207/f62ad8b4-5bdb-4142-b3ba-853fd10e234a)

Tabs switch
![Scherm­afbeelding 2024-03-21 om 15 28 24](https://github.com/nl-design-system/backlog/assets/545207/5914e46f-372e-4d7e-9417-121a781e6219)

---

### Notities
- Verschil tussen Figma en Storybook wat betreft responsive gedrag.
- Richtlijnen en beschrijving van keyboard navigatie beschikbaar (Storybook).

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/react-tabs--docs)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/Tab/src)
- [Figma - Huidig](https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?type=design&node-id=26905-136363&mode=design&t=DP3nkKq0PgXFaIFd-4)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-03-21 om 15 32 47](https://github.com/nl-design-system/backlog/assets/545207/d049d607-5510-4385-80ba-7e7b1d8a6e1d)

Update Tab Content
![Scherm­afbeelding 2024-03-21 om 15 32 52](https://github.com/nl-design-system/backlog/assets/545207/62e35160-ded8-46a1-b728-6dfb6b90af92)

Responsive gedrag bij small viewport
![Scherm­afbeelding 2024-03-21 om 17 26 24](https://github.com/nl-design-system/backlog/assets/545207/f2cc8d38-6ca6-4e98-902c-9f186e7ba515)


**Figma Huidig**

States
![Scherm­afbeelding 2024-03-21 om 17 23 30](https://github.com/nl-design-system/backlog/assets/545207/bd802ce9-c16f-4256-8a13-78dc06f33be4)

Responsive gedrag bij small viewport
![Scherm­afbeelding 2024-03-21 om 17 25 15](https://github.com/nl-design-system/backlog/assets/545207/efd11fa0-09e1-4679-ab90-50187e327dab)

Default en Fitted tabs
![Scherm­afbeelding 2024-03-21 om 17 25 32](https://github.com/nl-design-system/backlog/assets/545207/85cdbb3d-669b-4399-9bed-bca105668063)

---

### Notities
-  Richtlijnen beschikbaar.
- Verschil tussen Figma en Storybook wat betreft responsive gedrag.

**Comment by jeffreylauwers** [comment]
## DSO Toolkit

- [Documentatie website](https://dso-toolkit.nl/62.8.4/components/navigatie/tabs)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-03-25 om 08 37 44](https://github.com/nl-design-system/backlog/assets/545207/0c9794e1-b578-4ec5-a917-9e7cff50df61)

Inactive
![Scherm­afbeelding 2024-03-25 om 08 42 20](https://github.com/nl-design-system/backlog/assets/545207/ca58f137-14d5-4b20-b654-90f3022e8f3c)

Responsive gedrag bij small viewport
![Scherm­afbeelding 2024-03-25 om 08 45 12](https://github.com/nl-design-system/backlog/assets/545207/b556ddda-de70-4d31-aab4-faf60bc9713d)


---

### Notities

**Comment by jeffreylauwers** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/tabs)

---

### Screenshots / Varianten

**Documentatie website**

Page tabs
Gebruik een page tab als de inhoud van de tabs een groot deel van de pagina moet vormen.

![Scherm­afbeelding 2024-03-25 om 08 48 17](https://github.com/nl-design-system/backlog/assets/545207/125c1eb4-38a6-4e47-889a-78134e76eee5)

Responsive gedrag bij small viewport
![Scherm­afbeelding 2024-03-25 om 08 49 16](https://github.com/nl-design-system/backlog/assets/545207/d03b0c49-cd44-4f17-a001-73cd604291dc)



Inline tabs
Gebruik inline tabs als je de tabs tussen lopende tekst wil zetten, waarbij de inhoud van de tabs beknopt is en in verband staat met de rest van de content op de pagina.

![Scherm­afbeelding 2024-03-25 om 08 48 38](https://github.com/nl-design-system/backlog/assets/545207/bb37e1b8-fac0-4d38-9c28-f929828a59ed)

Responsive gedrag bij small viewport
![Scherm­afbeelding 2024-03-25 om 08 49 29](https://github.com/nl-design-system/backlog/assets/545207/0a9450cf-1759-479d-b19c-abb3933cb719)



Veel tabbladen
Als het gehele tab-element meer tabs bevat dan passen op de schermbreedte, verschijnt er een tab met de tekst “Meer”. Als de gebruiker op deze tab klikt, verschijnt er een dropdown met de tabs die niet meer op het scherm passen.

![Scherm­afbeelding 2024-03-25 om 08 49 54](https://github.com/nl-design-system/backlog/assets/545207/acc1e289-4cb7-45c2-991d-1a1468d30d95)

Responsive gedrag bij small viewport
![Scherm­afbeelding 2024-03-25 om 08 48 49](https://github.com/nl-design-system/backlog/assets/545207/07640527-050b-4618-9af6-e5cd05f2fe62)

![Scherm­afbeelding 2024-03-25 om 08 49 44](https://github.com/nl-design-system/backlog/assets/545207/bda0a934-0a51-4631-b065-78d628a1bbb2)

---

### Notities
- Richtlijnen beschikbaar.
- DUO heeft onderzoek gedaan naar het responsive gedrag van dit component. https://uno.dfront.rijkscloud.nl/#/over-uno/onderzoek-en-design/tabs

**Comment by jeffreylauwers** [comment]
## Open Gemeenten

- [Patternlab - TabbedTopTasks-List](https://patternlab.lenteveld.nl/patterns/02-organisms-050-Top-Tasks-30-Tabbed-Top-Tasks-OpenGemeentenTabbedTopTasks-List/02-organisms-050-Top-Tasks-30-Tabbed-Top-Tasks-OpenGemeentenTabbedTopTasks-List.html)
- [Documentatie website - Pagina accordeon](https://componenten.lenteveld.nl/pagina-accordeon)
- [Afspraken module](https://afspraak.barneveld.nl/datum-en-tijd/) (bijvoorbeeld bij maken van een afspraak voor ID kaart)

---

### Screenshots / Varianten

**Patternlab**

TabbedTopTasks-List
![Scherm­afbeelding 2024-03-25 om 08 57 45](https://github.com/nl-design-system/backlog/assets/545207/bacbeba5-4028-4868-b7ca-2b0b7a66cb00)

Responsive gedrag bij small viewport
![Scherm­afbeelding 2024-03-25 om 08 58 48](https://github.com/nl-design-system/backlog/assets/545207/027b8837-4282-4709-8667-d5604633cf70)

**Afspraken module**
![Scherm­afbeelding 2024-03-25 om 16 45 37](https://github.com/nl-design-system/backlog/assets/545207/c42970c4-93a7-496e-94d7-1a9c87582cfa)



---

### Notities

**Comment by jeffreylauwers** [comment]
## Rotterdam

[Figma](https://www.figma.com/file/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?type=design&node-id=19219-11650&mode=design&t=ZoRRQKdHcB9RgeLT-4)

---

### Screenshots / Varianten

**Figma**

![Scherm­afbeelding 2024-03-25 om 09 23 29](https://github.com/nl-design-system/backlog/assets/545207/e68b6a7d-a49e-419f-a292-533398fe179b)

Responsive gedrag bij small viewport
![Scherm­afbeelding 2024-03-25 om 09 23 36](https://github.com/nl-design-system/backlog/assets/545207/b9c771ca-9ad8-4051-ad18-523799e96d91)

![Scherm­afbeelding 2024-03-25 om 09 23 45](https://github.com/nl-design-system/backlog/assets/545207/f1ae03cf-c857-4534-934b-47bcd0a7633b)

---

### Notities
- Component is 'Under construction'.

**Comment by MariekeBrouwer** [comment]
![particulier](https://github.com/user-attachments/assets/5b92eb2f-07a9-4dff-a8dc-7a9f64c79d81)
![zakelijk](https://github.com/user-attachments/assets/0f0c4644-65f2-419d-8069-2c8654e933cb)

we gebruiken ook wel tabs voor verschillende inlogmogelijkheden. Bij het klikken op de tab verdwijnen de particulieren en gaat het dan naar de inlogmogelijkheden voor ondernemers. De tab zelf verkleurt dan en wordt dikker (ook vanwege toegankelijkheidseisen dat het niet enkel met kleur overgedragen mag worden).

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Tabs op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=tabs)

**Comment by rroose-rvo** [comment]
@Jeffrey Ik wil nog even terugkomen op het tabs onderwerp in de Open Hour van gisteren. Ik zag in jouw voorbeeld de subtiele versie van Tabs zoals ook wij die [momenteel gebruiken](https://nl-design-system.github.io/rvo/?path=/docs/componenten-tabs--documentatie). Uit onderzoek wat wij zelf gedaan hebben blijkt echter dat deze niet door elke gebruiker begrepen worden, ze lijken ook veel op normale links. Ik ben daarom bezig met [een minder subtiele versie](https://nl-design-system.github.io/rvo/iframe.html?args=&id=pagina-s-experimenteel-algemeen-tabs--default&viewMode=story) a la [gov.uk](https://design-system.service.gov.uk/components/tabs/), wellicht heb je er iets aan.

**Comment by DilemAliyazicioglu** [comment]
Bij RVO hebben we meerdere onderzoeken uitgevoerd waarin [tabs](https://nl-design-system.github.io/rvo/?path=/docs/componenten-tabs--documentatie) voorkwamen. In de meeste gevallen bleek dit component niet duidelijk voor alle testpersonen. In een onderzoek dat ik zelf heb uitgevoerd en geanalyseerd, begreep de helft van de testpersonen (4 personen) het component niet. Twee testpersonen ontdekten de tab pas later, en twee konden deze zonder hulp niet vinden.

Op basis daarvan (en op basis van eerdere onderzoeken van collega’s) concludeerden we het volgende: **de visuele vindbaarheid van een tabblad is cruciaal; wanneer een tab klein of subtiel is vormgegeven, wordt het gemakkelijk over het hoofd gezien.**

**Reply by rroose-rvo** [reply]
@DilemAliyazicioglu Ik heb dit intern uitgevraagd. Als ik reactie heb kom ik er hier op terug.

**Comment by eslook** [comment]
## ✨ Dit component is nu Community ✨


Help je mee hem door de Candidate stappen te krijgen?
[Tabs op het Candidate bord](https://github.com/orgs/nl-design-system/projects/80/views/1?filterQuery=tabs)
